### PR TITLE
Spec for StarkNet

### PR DESCRIPTION
**Pull request by delaaxe**
Adds CAIPs for StarkNet.

### CODE REVIEW

**Comment by bumblefudge** [comment]
More generally, you might think of CAIP10s and CAIP19s as "export formats" for addresses and assets in the system-- if there is a standardized and deterministic way to represent modulo-P/`felt` addresses and assets in modulo-2^256/"EVM-mode", that might be preferable to a lossful or arbitrary transformation of canonic representations

File: starknet/caip-2.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+*For context, see the [CAIP-2][] specification.*
+
+### Rationale / Syntax
+
+The namespace `starknet` refers to the StarkNet Layer 2 on Ethereum.
+
+StarkNet currently represents its chains with string identifiers such as 
+`SN_MAIN` and `SN_GOERLI`. They also have field element representations which
+are not worth discussing in this specification.
```
Comment (bumblefudge) [inline starknet/caip-2.md:24]:
```suggestion
`SN_MAIN` and `SN_GOERLI`. Internally, these strings are expressed as `field element` integers (according to a transformation documented in the system's [core documentation][Transaction structure]), but using the case-sensitive root string identifiers simplifies interoperability by sidestepping type system interoperability around integers.
```

File: starknet/caip-2.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# CAIP-2
+
+*For context, see the [CAIP-2][] specification.*
+
+### Rationale / Syntax
+
+The namespace `starknet` refers to the StarkNet Layer 2 on Ethereum.
+
+StarkNet currently represents its chains with string identifiers such as 
+`SN_MAIN` and `SN_GOERLI`. Internally, these strings are expressed as `field element` integers (according to a transformation documented in the system's [core documentation][Transaction structure]), but using the case-sensitive root string identifiers simplifies interoperability by sidestepping type system interoperability around integers.
```
Comment (iliav-starkware) [inline starknet/caip-2.md:23]:
```suggestion
`SN_MAIN` and `SN_GOERLI`. Internally, these strings are encoded as field elements (according to a transformation documented in the system's [core documentation][Transaction structure]), but using the case-sensitive root string identifiers simplifies interoperability by sidestepping type system interoperability around integers.
```

**Comment by delaaxe** [comment]
@bumblefudge I added a readme

File: starknet/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+This document describes the syntax and structure of the namespace for StarkNet..
+
+## Syntax
+
+The namespace "starknet" refers to the StarkNet Layer 2 chain on Ethereum.
+
+## References
+
+[Type system]: https://starknet.io/docs/reference/syntax.html#type-system
```
Comment (ArielElp) [inline starknet/README.md:25]: Cairo types seem like a very low-level link in this context, and will also stop being relevant with Cairo 1.0, so I think that we can remove this.

**Pull request merged**